Description: [PATCH] signal error on EOF in jpegload more reliably

 we were only warning on EOF in the read stub, even if fail was set

--- vips-8.10.5/libvips/foreign/jpeg2vips.c
+++ vips-8.10.5/libvips/foreign/jpeg2vips.c
@@ -212,6 +212,7 @@ typedef struct {
 
 	/* Private stuff during read.
 	 */
+	ReadJpeg *jpeg;
 	VipsSource *source;
 	unsigned char buf[SOURCE_BUFFER_SIZE];
 
@@ -248,7 +249,10 @@ source_fill_input_buffer( j_decompress_p
 		src->pub.bytes_in_buffer = read;
 	}
 	else {
-		WARNMS( cinfo, JWRN_JPEG_EOF );
+		if( src->jpeg->fail )
+			ERREXIT( cinfo, JERR_INPUT_EOF );
+		else
+			WARNMS( cinfo, JWRN_JPEG_EOF );
 		src->pub.next_input_byte = eoi_buffer;
 		src->pub.bytes_in_buffer = 2;
 	}
@@ -294,6 +298,7 @@ readjpeg_open_input( ReadJpeg *jpeg )
 				sizeof( Source ) );
 
 		src = (Source *) cinfo->src;
+		src->jpeg = jpeg;
 		src->source = jpeg->source;
 		src->pub.init_source = source_init_source;
 		src->pub.fill_input_buffer = source_fill_input_buffer;
